spec.html

```diff
@@ -781,26 +842,38 @@ <h1>ToElementDescriptor ( _elementObject_ )</h1>
       <emu-alg>
         1. Assert: _elementObject_ is an ECMAScript language value.
         1. Let _kind_ be ? ToString(? Get(_elementObject_, `"kind"`)).
-        1. If _kind_ is not one of `"method"` or `"field"`, throw a *TypeError* exception.
+        1. If _kind_ is not one of `"initializer"`, `"method"` or `"field"`, throw a *TypeError* exception.
         1. Let _key_ be ? Get(_elementObject_, `"key"`).
+        1. If _kind_ is `"initializer"`,
+          1. If _key_ is not *undefined*, throw a *TypeError* exception.
         1. If _key_ has a [[PrivateName]] internal slot, set _key_ to _key_.[[PrivateName]].
         1. Otherwise, set _key_ to ? ToPropertyKey(_key_).
         1. Let _placement_ be ? ToString(? Get(_elementObject_, `"placement"`)).
         1. If _placement_ is not one of `"static"`, `"prototype"`, or `"own"`, throw a *TypeError* exception.
-        1. Let _descriptor_ be ? ToPropertyDescriptor(? Get(_elementObject_, `"descriptor"`)).
-        1. Let _initializer_ be ? Get(_elementObject_, `"initializer"`).
-        1. Let _elements_ be ? Get(_elementObject_, `"elements"`).
-        1. If _elements_ is not *undefined*, throw a *TypeError* exception.
-        1. If _kind_ not `"field"`,
-          1. If _initializer_ is not *undefined*, throw a *TypeError* exception.
+        1. Let _descriptor_ be ? Get(_elementObject_, `"descriptor"`)
+        1. If _kind_ is `"initializer"`,
+          1. If _descriptor_ is not *undefined*, throw a *TypeError* exception.
+        1. Otherwise,
+          1. Set _descriptor_ to ? ToPropertyDescriptor(_descriptor_).
         1. If _key_ is a Private Name,
           1. If _descriptor_.[[Enumerable]] is *true*, throw a *TypeError* exception.
           1. If _descriptor_.[[Configurable]] is *true*, throw a *TypeError* exception.
           1. If _placement_ is `"prototype"`, throw a *TypeError* exception.
         1. If _kind_ is `"field"`,
           1. If _descriptor_ has a [[Get]], [[Set]] or [[Value]] internal slot, throw a *TypeError* exception.
-        1. Let _element_ be the ElementDescriptor { [[Kind]]: _kind_, [[Key]]: _key_, [[Placement]]: _placement_, [[Descriptor]]: _descriptor_ }.
-        1. If _kind_ is `"field"`, set _element_.[[Initializer]] to _initializer_.
+        1. Let _initializer_ be ? Get(_elementObject_, `"initializer"`).
+        1. If _kind_ is `"initializer"`,
+          1. If _initializer_ is *undefined*, throw a *TypeError* exception
+        1. If _kind_ is `"method"`,
+          1. If _initializer_ is not *undefined*, throw a *TypeError* exception.
+        1. Let _elements_ be ? Get(_elementObject_, `"elements"`).
+        1. If _elements_ is not *undefined*, throw a *TypeError* exception.
```

Comment:
Should this use a HasProperty, instead of a Get, since it’s just checking existence? Or is the intention to allow an explicit undefined property - and if so, that seems like it’d be an error, so why allow it?